server/settings_test.go

```diff
@@ -72,6 +73,188 @@ func TestSettings(t *testing.T) {
+				})
+				pmmapitests.AssertAPIErrorf(t, err, 400, codes.InvalidArgument, `Enable STT and disable STT cannot be both true`)
+				assert.Empty(t, res)
+			})
+
+			t.Run("EnableSTTAndEnableTelemetry", func(t *testing.T) {
+				defer teardown(t)
+
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableStt:       true,
+						EnableTelemetry: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.True(t, res.Payload.Settings.SttEnabled)
+				assert.True(t, res.Payload.Settings.TelemetryEnabled)
+				assert.Empty(t, err)
+			})
+
+			t.Run("EnableSTTAndDisableTelemetry", func(t *testing.T) {
+				defer teardown(t)
+
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableStt:        true,
+						DisableTelemetry: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				pmmapitests.AssertAPIErrorf(t, err, 400, codes.InvalidArgument, `Cannot enable STT while disabling telemetry`)
+				assert.Empty(t, res)
+			})
+
+			t.Run("DisableSTTAndEnableTelemetry", func(t *testing.T) {
+				defer teardown(t)
+
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						DisableStt:      true,
+						EnableTelemetry: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.False(t, res.Payload.Settings.SttEnabled)
+				assert.True(t, res.Payload.Settings.TelemetryEnabled)
+				assert.Empty(t, err)
+			})
+
+			t.Run("DisableSTTAndDisableTelemetry", func(t *testing.T) {
+				defer teardown(t)
+
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						DisableStt:       true,
+						DisableTelemetry: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.False(t, res.Payload.Settings.SttEnabled)
+				assert.False(t, res.Payload.Settings.TelemetryEnabled)
+				assert.Empty(t, err)
+			})
+
+			t.Run("EnableSTTWhileTelemetryEnabled", func(t *testing.T) {
+				defer teardown(t)
+
+				// Ensure Telemetry is enabled
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableTelemetry: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.True(t, res.Payload.Settings.TelemetryEnabled)
+
+				res, err = serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableStt: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+
+				assert.True(t, res.Payload.Settings.SttEnabled)
+				assert.Empty(t, err)
+
+				resg, err := serverClient.Default.Server.GetSettings(&server.GetSettingsParams{
```

Comment:
let's make checks like this (via GetSettings) everywhere we expect that settings should be updated